src/html5_parser/node/data/element.rs

```diff
@@ -45,8 +45,44 @@ impl ElementAttributes {
         self.attributes.contains_key(name)
     }
 
+    /// according to HTML5 spec: 3.2.3.1
+    /// https://www.w3.org/TR/2011/WD-html5-20110405/elements.html#the-id-attribute
+    fn validate_named_id(&self, named_id: &str) -> bool {
+        if named_id.contains(char::is_whitespace) {
+            return false;
+        }
+
+        if named_id.is_empty() {
+            return false;
+        }
+
+        // must contain at least one character, but
+        // doesn't specify it should *start* with a character
+        if !named_id.contains(char::is_alphabetic) {
+            return false;
+        }
+
+        true
+    }
```

Comment:
```suggestion
    fn validate_named_id(&self, named_id: &str) -> bool {
        if named_id.contains(char::is_whitespace) {
            return false;
        }

        if named_id.is_empty() {
            return false;
        }

        // must contain at least one character, but
        // doesn't specify it should *start* with a character
        named_id.contains(char::is_alphabetic)
    }
```

Reply:
(Sorry, remembering this PR is in draft right now.  Please ignore.)

Reply:
No worries, I like the suggestion